src/threshold-ts/tbtc/__test__/tbtc.test.ts

```
@@ -0,0 +1,739 @@
+          bitcoinAddressTestnet,
+          btcConfig.network
+        )
+        expect(isPublicKeyHashTypeAddress).toHaveBeenCalledWith(
+          bitcoinAddressTestnet
+        )
+        expect(calculateDepositRefundLocktime).toHaveBeenCalledWith(
+          Math.floor(mockCurrentTime / 1000),
+          23328000
+        )
+        expect(getChainIdentifier).toBeCalledWith(ethAddress)
+      })
+      test("should create proper deposit script parameters", async () => {
+        expect(depositScriptParameters.depositor).toBe(
+          getChainIdentifier(ethAddress)
+        )
+        expect(depositScriptParameters.blindingFactor).toBe(mockBlindingFactor)
+        expect(depositScriptParameters.refundPublicKeyHash).toBe(
+          mockDecodeBitcoinAddressResult
+        )
+        expect(depositScriptParameters.refundLocktime).toBe(
+          mockCalculateDepositRefundLocktimeResult
+        )
+        expect(depositScriptParameters.walletPublicKeyHash).toBe(
+          mockWalletPublicKeyHash
+        )
+      })
+    })
+
+    describe("when wrong btc address is provided", () => {
+      beforeEach(async () => {
+        setUpdeposiScriptParameters()
+        setUpValidBTCAddress(false)
+        setUpIsPublicKeyHashTypeAddress(false)
+      })
+      test("should throw", async () => {
+        await expect(
+          tBTC.createDepositScriptParameters(ethAddress, bitcoinAddressTestnet)
+        ).rejects.toThrow(
+          "Wrong bitcoin address passed to createDepositScriptParameters function"
+        )
+
+        expect(isValidBtcAddress).toHaveBeenCalledWith(
+          bitcoinAddressTestnet,
+          btcConfig.network
+        )
+      })
+    })
+
+    describe("when provided btc address is not a PKH type address", () => {
+      beforeEach(async () => {
+        setUpdeposiScriptParameters()
+        setUpValidBTCAddress(true)
+        setUpIsPublicKeyHashTypeAddress(false)
+      })
+      test("should throw", async () => {
```

maybe should throw an error?

---

7372eb86e267322da3686c50496aba647396c7ba